File: Medium/Java/Subsets(78).java
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<List<Integer>> forReturn = new ArrayList<List<Integer>>();  
        int index = 0, digit = 0;
        for (int i = 0; i < (int)Math.pow(2, nums.length); i++){
            digit = 0;
            index = i;
            List<Integer> toAdd = new ArrayList<Integer>();
            while(index > 1){
                if (index % 2 == 1){
                    toAdd.add(nums[digit]);
                }                
                index = index / 2;
                digit++;
            }
            if(index == 1) toAdd.add(nums[digit]);
            forReturn.add(toAdd);
        }
        return forReturn;
    }
}

File: Medium/Python 3/Subsets(78).py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        if len(nums) == 1:
            return [[],[nums[0]]]
        else:
            ans= [[], [nums[0]]]       
            for i in range(1,len(nums)):
                temp = len(ans)
                for j in range(temp):
                    ans.append(ans[j]+[nums[i]])  
            return ans